src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/WaitForBuildStepExecution.java

```diff
@@ -34,8 +36,18 @@ public boolean start() throws Exception {
             taskListener.getLogger().println("Waiting for " + runHyperLink + " to complete");
             return false;
         } else {
-            taskListener.getLogger().println(runHyperLink + " is already complete");
-            getContext().onSuccess(null);
+            Result result = run.getResult();
+            if (result == null) {
```

Comment:
this should never occur (for a [pipeline job](https://github.com/jenkinsci/workflow-job-plugin/blob/d2290d3341af6011b1f7833ab57d4db28b824e3a/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L797-L799C6) or a [FreeStyle](https://github.com/jenkinsci/jenkins/blob/019950a4a302db7f1094d49be9025866307356d5/core/src/main/java/hudson/model/Run.java#L527-L534) job due to its [state](https://github.com/jenkinsci/jenkins/blob/019950a4a302db7f1094d49be9025866307356d5/core/src/main/java/hudson/model/Run.java#L266-L275),   as such I would this should at least be shown in the logs that the build is in an indeterminate state

```suggestion
            if (result == null) {
                taskListener.getLogger().println("Warning: " + runHyperLink + " already completed but getResult() returned null.  Treating the result of this build as a failure));
```